src/uk/zenko/src/eu/kanade/tachiyomi/extension/uk/zenko/Zenko.kt

```diff
@@ -0,0 +1,199 @@
+        return GET(url, headers)
+    }
+
+    override fun pageListParse(response: Response): List<Page> {
+        val data = response.asClass<ChapterResponseItem>()
+        return data.pages!!.map { page ->
+            Page(page.id, "", "$IMAGE_STORAGE_URL/${page.imgUrl}")
+        }
+    }
+
+    override fun imageUrlParse(response: Response): String = ""
+
+    // ============================= Utilities ==============================
+    private fun parseAsMangaResponseDto(response: Response): MangasPage {
+        val zenkoMangaListResponse = response.asClass<ZenkoMangaListResponse>()
+        return makeMangasPage(zenkoMangaListResponse.data, zenkoMangaListResponse.meta.hasNextPage)
+    }
+
+    private fun offsetCounter(page: Int): Int {
+        var currentOfsset = 0
+        val offset = if (page == 1) {
+            currentOfsset = 0 // for page 1 offset must be 0
+            currentOfsset
+        } else {
+            currentOfsset = (page - 1) * 15 // calculate offset for other pages
+            currentOfsset
+        }
+        return offset
+    }
+
+    private fun makeZenkoMangaRequest(offset: Int, sortBy: String): Request {
+        val url = "$API_URL/titles".toHttpUrl().newBuilder()
+            .addQueryParameter("limit", "15")
+            .addQueryParameter("offset", "$offset")
+            .addQueryParameter("sortBy", sortBy)
+            .addQueryParameter("order", "DESC")
+            .build()
+        return GET(url, headers)
+    }
+
+    private fun makeMangasPage(
+        mangaList: List<MangaDetailsResponse>?,
+        hasNextPage: Boolean = false,
+    ): MangasPage {
+        return MangasPage(
+            mangaList!!.map(::makeSManga),
+            hasNextPage,
+        )
+    }
+
+    private fun makeSManga(mangaDto: MangaDetailsResponse) = SManga.create().apply {
+        title = mangaDto.engName ?: mangaDto.name
+        thumbnail_url =
+            "$IMAGE_STORAGE_URL/${mangaDto.coverImg}?optimizer=image&width=560&quality=70&height=auto"
+        url = "$baseUrl/titles/${mangaDto.id}"
+        status = mangaDto.status.toStatus()
+    }
+
+    private fun String.toStatus(): Int {
+        return when (this) {
+            "ONGOING" -> SManga.ONGOING
+            "FINISHED" -> SManga.COMPLETED
+            "PAUSED" -> SManga.ON_HIATUS
+            else -> SManga.UNKNOWN
+        }
+    }
+
+    private fun Long.secToMs(): Long {
+        return (
+            runCatching { this * 1000 }
+                .getOrNull() ?: 0L
+            )
+    }
+
+    companion object {
+        private const val API_URL = "https://zenko-api.onrender.com"
+        private const val IMAGE_STORAGE_URL = "https://zenko.b-cdn.net"
+
+        private val json = Json {
+            ignoreUnknownKeys = true
+            coerceInputValues = true
+        }
+
+        private inline fun <reified T> Response.asClass(): T = use {
+            json.decodeFromStream(it.body.byteStream())
+        }
```

Comment:
```suggestion
        private inline fun <reified T> Response.parseAs(): T  {
            json.decodeFromStream(body.byteStream())
        }
```
nitpick on function name, and take it out of the companion object